packages/calypso-e2e/src/lib/pages/reader-page.ts

```diff
@@ -28,54 +28,42 @@ export class ReaderPage {
 		this.page = page;
 	}
 
-	/**
-	 * Verifies that we are actually on the Reader Page
-	 *
-	 * @returns {Promise<void>} No return value.
-	 */
-	private async verifyReaderPage(): Promise< void > {
-		await this.page.waitForSelector( selectors.readerPageLocator );
-	}
-
 	/**
 	 * Get the URL of the latest post in Reader
 	 *
 	 * @returns {Promise<string>} String of URL for latest post.
 	 */
 	async siteOfLatestPost(): Promise< string > {
-		const href = await this.page.getAttribute( selectors.readerVisitLink, 'href' );
+		const href = await this.page.getAttribute( selectors.visitSiteLink, 'href' );
 		return new URL( href ? href : '' ).host;
 	}
 
 	/**
-	 * Share latest post in Reader
+	 * Visits a post in the Reader.
 	 *
-	 * @returns {Promise<void>} No return value.
+	 * @param param0 Keyed object parameter.
+	 * @param {number} param0.index n-th post to view on the reader page. 1-indexed.
 	 */
-	async shareLatestPost(): Promise< void > {
-		await this.page.click( selectors.shareButtonLocator );
-		await this.page.click( selectors.siteContentLocator );
-	}
+	async visitPost( { index }: { index?: number } = {} ): Promise< void > {
+		let selector = '';
+		if ( index ) {
+			selector = `:nth-match(${ selectors.readerCard }, ${ index })`;
+		}
 
-	/**
-	 * Sets and submits comment on latest post
-	 *
-	 * @param {string} comment Text of the comment.
-	 * @returns {Promise<void>} No return value.
-	 */
-	async commentOnLatestPost( comment: string ): Promise< void > {
-		await this.page.click( selectors.commentButtonLocator );
-		await this.page.fill( selectors.commentTextAreaLocator, comment );
-		await this.page.click( selectors.commentSubmitLocator );
+		await Promise.all( [ this.page.waitForNavigation(), this.page.click( selector ) ] );
 	}
 
 	/**
-	 * Wait for the comment with the specified text to display.
+	 * Sets and submits comment on latest post
 	 *
 	 * @param {string} comment Text of the comment.
 	 * @returns {Promise<void>} No return value.
 	 */
-	async waitForCommentToAppear( comment: string ): Promise< void > {
+	async comment( comment: string ): Promise< void > {
```

Comment:
Totally optional, but do you think we should change this method name to indicate that this applies to a visited post? Like in this flow, you have to visit the post first, then you can call this method. So something like `commentOnVisitedPost`?

Reply:
I'm not sure if renaming the method to a more specific `commentOnVisitedPost` is required here. 

If the Reader did support leaving comments inline, then I can see this being a source of confusion, but it appears that user would need to visit a post in order to leave a comment. Also, this subsequent part is a stylistic matter so it is subjective, but I tend to prefer more generic names where possible.